lib/engine/game/g_1848/game.rb

```diff
@@ -249,21 +267,29 @@ class Game < Game::Base
             desc: 'Owning Public Company or its Director may build one (1) free tile on a desert hex (marked by'\
                   ' a cactus icon). This power does not go away after a 5/5+ train is purchased.',
             abilities: [
-              {
-                type: 'tile_discount',
-                discount: 40,
-                terrain: 'desert',
-                count: 1,
-                owner_type: 'corporation',
-              },
-              {
-                type: 'tile_discount',
-                discount: 40,
-                terrain: 'desert',
-                count: 1,
-                owner_type: 'player',
-              },
-            ],
+                    {
+                      type: 'tile_lay',
+                      discount: 40,
+                      hexes: %w[B7 B9 C2 C4 C8 E6 E8],
+                      tiles: %w[7 8 9],
+                      count: 1,
+                      reachable: true,
+                      consume_tile_lay: true,
+                      owner_type: 'corporation',
+                      when: 'owning_corp_or_turn',
+                    },
+                    {
+                      type: 'tile_lay',
+                      discount: 40,
+                      hexes: %w[B7 B9 C2 C4 C8 E6 E8],
+                      tiles: %w[7 8 9],
+                      count: 1,
+                      reachable: true,
+                      consume_tile_lay: true,
+                      owner_type: 'player',
+                      when: 'owning_player_or_turn',
+                    },
+                  ],
```

Comment:
FYI when one of the abilities is used, you'll need a way to remove the other (or set the count to 0).

Reply:
I'm aware, i put it in the do list.
My plan of attack is make the privates work, then fix the edge cases